applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -190,37 +193,46 @@ K_WORK_DEFINE(cap_start_work, cap_start_worker);
  * @retval	0	Operation successful.
  * @retval	-EINVAL	Any error.
  */
-static int device_pres_delay_find(uint8_t index, uint32_t *pres_dly_us)
+static int device_pres_delay_find(struct stream_index idx, uint32_t *pres_dly_us)
 {
-	uint32_t pd_min = unicast_servers[index].sink_ep->qos_pref.pd_min;
-	uint32_t pd_max = unicast_servers[index].sink_ep->qos_pref.pd_max;
-	uint32_t pref_pd_min = unicast_servers[index].sink_ep->qos_pref.pref_pd_min;
-	uint32_t pref_pd_max = unicast_servers[index].sink_ep->qos_pref.pref_pd_max;
+	struct le_audio_unicast_server *unicast_server =
+		&unicast_servers[idx.lvl1][idx.lvl2][idx.lvl3];
 
-	LOG_DBG("Index: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", index,
+	uint32_t pd_min = unicast_server->sink_ep->qos_pref.pd_min;
+	uint32_t pd_max = unicast_server->sink_ep->qos_pref.pd_max;
+	uint32_t pref_pd_min = unicast_server->sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_pd_max = unicast_server->sink_ep->qos_pref.pref_pd_max;
+
+	LOG_DBG("Index: %d, Pref min: %d, pref max: %d, pres_min: %d, pres_max: %d", idx.lvl3,
 		pref_pd_min, pref_pd_max, pd_min, pd_max);
 
 	*pres_dly_us = 0;
 
-	for (int i = 0; i < ARRAY_SIZE(unicast_servers); i++) {
-		if (le_audio_ep_qos_configured(unicast_servers[i].sink_ep)) {
+	for (int i = 0; i < ARRAY_SIZE(unicast_servers[idx.lvl1][idx.lvl2]); i++) {
```

Comment:
Consider adding `unicast_servers[idx.lvl1][idx.lvl2]` as a variable here as well - it's accessed 11 places in the loop